addons/analytic/models/analytic_plan.py

```diff
@@ -250,14 +260,46 @@ def _get_applicability(self, **kwargs):
     def unlink(self):
         # Remove the dynamic field created with the plan (see `_inverse_name`)
         self._find_plan_column().unlink()
-        return super().unlink()
+        related_fields = self._find_related_field()
+        res = super().unlink()
+        related_fields.filtered(lambda f: not self._is_subplan_field_used(f)).unlink()
+        self.env.registry.clear_cache()
+        return res
+
+    def _hierarchy_name(self):
+        depth = self.parent_path.count('/') - 1
+        fname = f"{self._column_name()}_{depth}"
+        if fname.startswith('account_id'):
+            fname = f'x_{fname}'
+        return depth, fname
+
+    def _is_subplan_field_used(self, field):
+        """Return `True` if there are analytic plans still on the same hierarchy level as what the field was created for.
+
+        :param field: the recordset of a field created to group by sub plan
+        :rtype: bool
+        """
+        assert '_id_' in field.name
+        root_name, depth = field.name.rsplit('_', maxsplit=1)
+        plan_id_match = re.search(r'\d+', root_name)
+        plan_id = int(plan_id_match.group() if plan_id_match else self._get_all_plans()[0])
```

Comment:
This `self._get_all_plans()[0])` returns a traceback as map is not subscriptable.
`next` I suppose?

(create a subplan to Project. Delete it)